src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,108 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_CAPACITY = 16;
+    private Node<K, V>[] array;
+    private int threshhold;
+    private int size;
+
+    public MyHashMap() {
+        array = (Node<K,V>[]) new Node[DEFAULT_CAPACITY];
+        threshhold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
+
+    @Override
+    public int getSize() {
+        return size;
+    }
+
+    private int hash(K key) {
+        int h;
+        return (key == null) ? 0 : (h = key.hashCode()) ^ (h >> 16);
+    }
 
     @Override
     public void put(K key, V value) {
+        int keyHash = hash(key);
+        int index = keyHash % array.length;
+        Node<K, V> node = new Node<>(keyHash, key, value, null);
+        put(array, node, index);
+        if (size > threshhold) {
+            resize();
+        }
+    }
+
+    private void put(Node<K, V>[] toArray, Node<K, V> newNode, int index) {
+        if (toArray[index] == null) {
+            toArray[index] = newNode;
+            size++;
+            return;
+        }
+        Node<K, V> node = toArray[index];
+        while (!Objects.equals(node.key, newNode.key) && node.next != null) {
+            node = node.next;
+        }
+        if (Objects.equals(newNode.key, node.key)) {
+            node.value = newNode.value;
+            return;
+        }
+        node.next = newNode;
+        size++;
+    }
 
+    private void resize() {
+        int newCapacity = array.length * 2;
+        int index;
+        size = 0;
+        threshhold = (int) (newCapacity * DEFAULT_LOAD_FACTOR);
+        Node<K, V>[] newArray = (Node<K,V>[]) new Node[newCapacity];
+        for (Node<K, V> node : array) {
+            while (node != null) {
+                index = node.hash % newArray.length;
```

Comment:
```suggestion
        size = 0;
        threshhold = (int) (newCapacity * DEFAULT_LOAD_FACTOR);
        Node<K, V>[] newArray = (Node<K,V>[]) new Node[newCapacity];
        for (Node<K, V> node : array) {
            while (node != null) {
                int index = node.hash % newArray.length;
```
we don't use index outside the loop, let's declare it there